sdk/src/lib/ik-image/ik-image.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, AfterViewInit, OnInit, ElementRef, Input, OnChanges } from '@angular/core';
+import { ImagekitService } from '../imagekit-angular.service';
+import { Dict, QueryParameters, IkImageComponentOptions } from '../utility/ik-type-def-collection';
+import { getInt } from '../utility/utils';
+import { buildSrc, getResponsiveImageAttributes, SrcOptions } from '@imagekit/javascript';
+import type { Transformation } from '@imagekit/javascript'
+
+@Component({
+  selector: 'ik-image',
+  template: `<img src='' (load)="onImageLoaded($event)">`,
+})
+export class IkImageComponent implements AfterViewInit, OnInit, OnChanges {
+  @Input('src') src: string = "";
+  @Input('urlEndpoint') urlEndpoint: string;
+  @Input('transformation') transformation: Array<Transformation> = [];
+  @Input('transformationPosition') transformationPosition: "path" | "query";
+  @Input('queryParameters') queryParameters: QueryParameters;
+  @Input('loading') loading: string = "lazy";
+  @Input('width') width: number | string;
+  @Input('height') height: string;
+  @Input('responsive') responsive: boolean = true;
+  @Input('deviceBreakpoints') deviceBreakpoints: number[];
+  @Input('imageBreakpoints') imageBreakpoints: number[];
+  @Input('sizes') sizes: string;
+  url = '';
+  srcset = '';
+
+  constructor(private el: ElementRef, private imagekit: ImagekitService) {
+  }
+
+  ngOnInit(): void {
+    const options: IkImageComponentOptions = {
+      src: this.src,
+      urlEndpoint: this.urlEndpoint ? this.urlEndpoint : this.imagekit._ikInstance.options.urlEndpoint,
+      transformation: this.transformation,
+      transformationPosition: this.transformationPosition,
+      queryParameters: this.queryParameters,
+    };
+    this.setUrl(options);
+  }
+
+  ngOnChanges(): void {
```

Comment:
Calling lifecycle hooks manually is anti-pattern. Instead, extract the logic in these hooks into separate functions and then call them from both the places.

Ref: https://www.linkedin.com/pulse/why-you-shouldnt-trigger-lifecycle-hooks-manually-prakhar-srivastava-ybxnf/